agent/reconciler/host_reconciler.go

```diff
@@ -156,6 +166,33 @@ func (r *HostReconciler) reconcileNormal(ctx context.Context, byoHost *infrastru
 	return ctrl.Result{}, nil
 }
 
+func (r *HostReconciler) executeInstallerController(ctx context.Context, byoHost *infrastructurev1beta1.ByoHost) error {
+	logger := ctrl.LoggerFrom(ctx)
+	secret := &corev1.Secret{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: byoHost.Spec.InstallationSecret.Name, Namespace: byoHost.Spec.InstallationSecret.Namespace}, secret)
+	if err != nil {
+		logger.Error(err, "error getting install and uninstall script")
+		r.Recorder.Eventf(byoHost, corev1.EventTypeWarning, "ReadInstallationSecretFailed", "install and uninstall script %s not found", byoHost.Spec.InstallationSecret.Name)
+		return err
+	}
+	installScript := string(secret.Data["install"])
+	uninstallScript := string(secret.Data["uninstall"])
+
+	byoHost.Spec.UninstallationScript = &uninstallScript
+	installScript, err = r.parseScript(ctx, installScript)
+	if err != nil {
+		return err
+	}
+	logger.Info("executing install script")
+	err = r.executeScript(ctx, installScript)
+	if err != nil {
+		logger.Error(err, "error executing installation script")
+		r.Recorder.Event(byoHost, corev1.EventTypeWarning, "InstallScriptExecutionFailed", "install script execution failed")
+		return err
+	}
```

Comment:
in old installation flow, when installation fails, we are setting
```
logger.Error(err, "error in installing k8s components")
r.Recorder.Event(byoHost, corev1.EventTypeWarning, "InstallK8sComponentFailed", "k8s component installation failed")
conditions.MarkFalse(byoHost, infrastructurev1beta1.K8sComponentsInstallationSucceeded, infrastructurev1beta1.K8sComponentsInstallationFailedReason, clusterv1.ConditionSeverityInfo, "")
```

Can we do the same in this new installation flow as well?

Reply:
I have added this condition in the block. Thanks